src/main/java/tc/oc/pgm/listeners/ServerPingDataListener.java

```diff
@@ -0,0 +1,154 @@
+package tc.oc.pgm.listeners;
+
+import static com.google.common.base.Preconditions.*;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.server.ServerListPingEvent;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchManager;
+import tc.oc.pgm.api.match.event.MatchLoadEvent;
+import tc.oc.pgm.map.Contributor;
+import tc.oc.pgm.map.MapInfo;
+import tc.oc.pgm.map.PGMMap;
+import tc.oc.pgm.rotation.PGMMapOrder;
+import tc.oc.util.logging.ClassLogger;
+
+public class ServerPingDataListener implements Listener {
+
+  private final MatchManager matchManager;
+  private final Logger logger;
+  private final AtomicBoolean ready;
+  private final LoadingCache<Match, JsonObject> matchCache;
+
+  public ServerPingDataListener(MatchManager matchManager, Logger parentLogger) {
+    this.matchManager = checkNotNull(matchManager);
+    this.logger = ClassLogger.get(checkNotNull(parentLogger), ServerPingDataListener.class);
+    this.ready = new AtomicBoolean();
+    this.matchCache =
+        CacheBuilder.newBuilder()
+            .weakKeys()
+            .expireAfterWrite(5L, TimeUnit.SECONDS)
+            .build(
+                new CacheLoader<Match, JsonObject>() {
+                  @Override
+                  public JsonObject load(Match match) throws Exception {
+                    JsonObject jsonObject = new JsonObject();
+                    serializeMatch(match, jsonObject);
+                    return jsonObject;
+                  }
+                });
+  }
+
+  @EventHandler
+  public void onMatchLoad(MatchLoadEvent event) {
+    ready.compareAndSet(false, true);
+  }
+
+  @EventHandler
+  public void onServerListPing(ServerListPingEvent event) {
+    if (!ready.get()) return;
+
+    JsonObject root = event.getOrCreateExtra(PGM.get());
```

Comment:
Can't you just try/catch this for the method not found exception/throwable, and return if it throws? AFAIK it's the only thing that uses a new sportbukkit method